contracts/math/uints.ral

```diff
@@ -90,5 +90,52 @@ Abstract Contract U512Impl () {
                 lower: deltaL
             }
         }
+
+    pub fn bigAdd512(a: U512, b: U512) -> U512 {
+        let (r0, overflow) = overflowingAdd256(a.lower, b.lower)
+        let r1 = wrappingAdd256(wrappingAdd256(a.higher, b.higher), overflow)
+        return U512 { higher: r1, lower: r0 }
+    }
+
+    pub fn bigMul256(a: U256, b: U256) -> U512 {
```

Comment:
will `bigMulUp` will be added within this PR?

Reply:
Have you tried to optimalize it? GPT could be usefull

Reply:
+1